src/stores/calenderizeHelpers.ts

```diff
@@ -2,10 +2,10 @@ import { CourseEvent, CustomEvent } from '../components/Calendar/CourseCalendarE
 import AppStore from './AppStore';
 
 export const calendarizeCourseEvents = () => {
```

Comment:
```suggestion
export const calendarizeCourseEvents = (currentCourses: CourseEvent[]) => {
```
I personally think this would make the helper functions perfectly decoupled; there's this weird circular dependency between them and the AppStore: helpers ask the AppStore for data and the AppStore asks the helpers for data;

By just passing in `currentCourses` as a param, the `AppStore.schedule.getCurrentCourses` can easily be called within the AppStore itself, helper functions would become independent of the AppStore's implementation, and the circular dependency is resolved.

This does require that you fix the function signature of each helper function to ask for any data it needs from AppStore as params, and then adjust the calls from AppStore itself. I think it would be a good improvement, but I'll also address it during my own re-factoring work

Reply:
At some point I was going to move the calenderizing functions into `Schedule.ts` but I never got around to it. Will probably add it in an upcoming commit.

Reply:
Decided `Schedules.ts` becomes a little too bloated, changes were implemented.